application/F3DStarter.cxx

```diff
@@ -858,7 +858,27 @@ class F3DStarter::F3DInternals
       f3d::window& window = this->Engine->getWindow();
       if (this->AppOptions.Resolution.size() == 2)
       {
-        window.setSize(this->AppOptions.Resolution[0], this->AppOptions.Resolution[1]);
+        F3DOptionsTools::OptionsDict appOptions = F3DOptionsTools::DefaultAppOptions;
+        std::vector<int> defaultRes;
+
+        this->ParseOption(appOptions, "resolution", defaultRes);
+
+        // If user did not set window size, scale it base on fontScale and system zoom scale
+        if (defaultRes != this->AppOptions.Resolution)
```

Comment:
what if user set resolution to default res ? That doesnt work Im afraid. You need an optional here instead.